File: 베스트앨범.js
function solution(genres, plays) {
    var answer = [];
    let dict = {};
    let genres_dict = {};

    for (var i in genres) {
        if (genres[i] in dict) {
            dict[genres[i]] += plays[i];
            genres_dict[genres[i]].push([plays[i], parseInt(i)]);
        } else {
            dict[genres[i]] = plays[i];
            genres_dict[genres[i]] = [[plays[i], parseInt(i)]];
        }
    }

    let arr = [];

    for (var i in dict) {
        arr.push([dict[i], i]);
        genres_dict[i].sort(function (a, b) {
            var tmp = b[0] - a[0];
            if (tmp == 0) {
                return a[1] - b[1];
            } else {
                return tmp;
            }
        });
    }

    arr.sort(function (a, b) {
        return b[0] - a[0];
    });

    for (var i in arr) {
        var now_genre = arr[i][1];
        for (var j = 0; j < Math.min(2, genres_dict[now_genre].length); j++) {
            answer.push(genres_dict[now_genre][j][1]);
        }
    }

    return answer;
}

File: 베스트앨범.py
from collections import defaultdict

def solution(genres, plays):
    answer = []
    
    genre_dict = defaultdict(lambda: 0)
    total_list = defaultdict(lambda: [])

    for i in range(0, len(genres)):
        genre_dict[genres[i]] += plays[i]
        total_list[genres[i]].append((plays[i], i))

    genre_dict = sorted(genre_dict.items(), key=lambda x:x[1], reverse=True)

    for i in total_list.keys():
        total_list[i] = sorted(total_list[i], key=lambda x:x[0], reverse=True)

    for i in genre_dict:
        if len(total_list[i[0]]) > 1:
            answer.append(total_list[i[0]][0][1])
            answer.append(total_list[i[0]][1][1])
        else:
            answer.append(total_list[i[0]][0][1])

    return answer
